From 8e1b984f07636e10c0ccf8beed79a66112001a30 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 19 Sep 2024 18:23:21 +0200
Subject: [PATCH 1242/1695] plat-stm32mp2: enable all OPP support for M33TDCID

Return true in stm32mp_supports_cpu_opp() in M33TDCID as we do not have
access to OTP and stm32_bsec3 driver does not read the bsec mirror.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git bb028762447e46da6175dadc4660a88e3d8a40be plat-stm32mp2: enable all OPP support for M33TDCID]
Change-Id: Id43e1ddcde7364648cc14118767abec74dc1260c
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/405846
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/plat-stm32mp2/main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/main.c b/core/arch/arm/plat-stm32mp2/main.c
index c9e3812fc..aec3929c8 100644
--- a/core/arch/arm/plat-stm32mp2/main.c
+++ b/core/arch/arm/plat-stm32mp2/main.c
@@ -257,6 +257,9 @@ bool stm32mp_supports_cpu_opp(uint32_t opp_id __maybe_unused)
 		id = BIT(0);
 
 	return (opp_id & id) == id;
+#elif CFG_STM32_CM33TDCID
+	/* can't read OTP in M33TDCID to support all OPP */
+	return true;
 #else
 	return false;
 #endif
-- 
2.47.2

